R/get-param.R

```diff
@@ -0,0 +1,155 @@
+
+#' Return a formatted object for a given parameter
+#'
+#' @param .summary a `bbi_nonmem_summary` or `bbi_summary_list` object.
+#' @param .param the parameter(s) to retrieve and format. Any subset of `c("omega", "sigma", "theta")`
+#'
+#' @importFrom stats setNames
+#' @keywords internal
+get_param <- function(.summary, .param = c("omega", "sigma", "theta")){
+
+  if (inherits(.summary, NM_SUM_CLASS)) {
+    .summary <- list(.summary)
+    check_model_object_list(.summary, .mod_types = c(NM_SUM_CLASS))
+  }else if(inherits(.summary, SL_CLASS)){
+    check_model_object(.summary, .mod_types = c(SL_CLASS))
+  }else{
+    stop("Please pass in a `", NM_SUM_CLASS, "` or `", SL_CLASS, "` type object.")
+  }
+
+  assert_true(all(.param %in% c("omega", "sigma", "theta")))
+
+  param_obj <- map(.param, function(param.i){
+
+    if(param.i %in% c("omega", "sigma")){
+      param_obj_i <- map(.summary, ~ {
+        # unpack bbi_summary_list element
+        if (!is.null(.x$bbi_summary)) .x <- .x$bbi_summary
+
+        param_names <- .x[[SUMMARY_PARAM_NAMES]]
+        estimates <- .x[[SUMMARY_PARAM_DATA]][[length(.x[[SUMMARY_PARAM_DATA]])]]$estimates
+        format_matrix(estimates[[param.i]], param_names[[param.i]], .type = toupper(param.i))
+      })
+
+    }else if(param.i %in% c("theta")){
+      param_obj_i <- map(.summary, ~ {
+        # unpack bbi_summary_list element
+        if (!is.null(.x$bbi_summary)) .x <- .x$bbi_summary
+
+        param_names <- .x[[SUMMARY_PARAM_NAMES]]
+        theta <- .x[[SUMMARY_PARAM_DATA]][[length(.x[[SUMMARY_PARAM_DATA]])]]$estimates$theta
+        names(theta) <- param_names$theta
+        theta
+      })
+    }
+
+    if(length(param_obj_i) == 1) param_obj_i <- param_obj_i[[1]]
+    param_obj_i
+  }) %>% setNames(.param)
+
+  return(param_obj)
+}
+
+
+
+#' Return a formatted object for a given parameter
+#'
+#' @description
+#'
+#' * [get_omega()]: Return omega values as a labeled symmetric matrix
+#' * [get_sigma()]: Return sigma values as a labeled symmetric matrix
+#' * [get_theta()]: Return theta values as a named vector
+#'
+#' @param .summary a `bbi_nonmem_summary` or `bbi_summary_list` object.
+#'
+#' @examples
+#'
+#' \dontrun{
+#' sum <- mod %>% model_summary()
+#'
+#'
+#' sum %>% get_omega()
+#'               OMEGA_1     OMEGA_2     OMEGA_3      OMEGA_4
+#'  OMEGA_1  0.010410500 0.000319542 0.001977200 -0.000189864
+#'  OMEGA_2  0.000319542 0.008278020 0.000872168  0.001005720
+#'  OMEGA_3  0.001977200 0.000872168 0.013139700  0.004040370
+#'  OMEGA_4 -0.000189864 0.001005720 0.004040370  0.010987300
+#'
+#'
+#' sum %>% get_sigma()
+#'          SIGMA_1
+#'  SIGMA_1       1
+#'
+#'
+#' sum %>% get_theta()
+#'     THETA1     THETA2     THETA3     THETA4     THETA5     THETA6     THETA7     THETA8     THETA9    THETA10    THETA11
+#'  3.3005000  3.2519700 -0.6122190 -0.2080120  0.7372940  1.1359900  0.3346290  0.1923520  0.6916970  2.2993700  0.0991684
```

Comment:
This comes triggers a note:

```
❯ checking Rd line widths ... NOTE
  Rd file 'get_omega.Rd':
    \examples lines wider than 100 characters:
           THETA1     THETA2     THETA3     THETA4     THETA5     THETA6     THETA7     THETA8     THETA9    THETA10    THETA11
        3.3005000  3.2519700 -0.6122190 -0.2080120  0.7372940  1.1359900  0.3346290  0.1923520  0.6916970  2.2993700  0.0991684
```

That's not really a problem, though it is nice to keep the check output as quiet as possible.  Maybe it'd be enough to just show, say, the `get_omega()` calls.  An example is nice, but once you've seen one, I'm not sure seeing each other one adds much value.  Up to you, just a thought.